### PR TITLE
sound: edit and restructure sound module pull request

### PR DESCRIPTION
**Pull request by JaStarbuck**
Summary: I have finished generalizing the structure of the demo into a load, play, and free functions. This was done by looking at demo.c and understanding the thought process and splitting it up into new functions in sound.c that were declared in sound.h. Hopefully these functions will be able to be used by other teams and used to integrate sound into games made in chiventure. After the fact, we have some warnings that need to be worked on in a new issue and in the future, we can create more functions including stop/pause that can be used for increased functionality.

First comment: I'm having trouble with compilation because of include statement errors and building chiventure in the workflow. However, the functions I wrote don't look like theres anything glaringly wrong with them. Maybe we could pull it into dev so others can look and we can debug.

### CODE REVIEW

**Comment by JaStarbuck** [comment]
Debugged based off of workflow, got all tests to pass #1319

**Review by parkerbroz — APPROVED** [review]
Great documentation with the comments - made code easy to understand and review. Your implementation seems intuitive and functional. I think our team and other teams will have an easy time working with this module in the future.

File: src/sound/src/sound.c [hunk 1]
```diff
@@ -5,17 +5,62 @@
 /* See sound.h for details */
 int load_wav(sound_t *sound)
 {
+    //This initializes the all SDL2 assets
+    SDL_Init(SDL_INIT_EVERYTHING | SDL_INIT_VIDEO); 
+
+    //Basic Audio Format, future teams may want to experiment with this
+    Mix_OpenAudio(44100, MIX_DEFAULT_FORMAT,2, 2048);
+
+    if (sound == NULL){ // error check, nothing passed into function
```
Comment (ecoble) [inline src/sound/src/sound.c]: Please add a space between ){, you can check the [style guide](https://uchicago-cs.github.io/dev-guide/style_guide_c.html#style-guide-c) and run a style to automatically fix style issues.

File: src/sound/src/sound.c [hunk 2]
```diff
@@ -5,17 +5,62 @@
...
+        fprintf(stderr, "no sound objects present\n");
+        exit(1);
+    } else if (sound->type == BACKGROUND) {
+        Mix_Music *mus = Mix_LoadMUS(sound->name); // loads the music to be played
+        if (!mus){ // error check
+            printf("Background music is not loaded\n");
+            return 1;
+        }
+    } else if (sound->type == SOUND_EFFECT){
```
Comment (ecoble) [inline src/sound/src/sound.c]: Same here with style.

File: src/sound/src/sound.c [hunk 3]
```diff
@@ -5,17 +5,62 @@
...
+        exit(1);
+    } else if (sound->type == BACKGROUND) {
+        Mix_Music *mus = Mix_LoadMUS(sound->name); // loads the music to be played
+        if (!mus){ // error check
+            printf("Background music is not loaded\n");
+            return 1;
+        }
+    } else if (sound->type == SOUND_EFFECT){ 
+        Mix_Chunk *wav = Mix_LoadWAV(sound->name); // loads the sound effect to be played
+        if (!wav){ // error check
```
Comment (ecoble) [inline src/sound/src/sound.c]: Same here with style.

File: src/sound/src/sound.c [hunk 4]
```diff
@@ -5,17 +5,62 @@
...
+            return 1;
+        }
+    }
     return 0;
 }
 
 /* See sound.h for details */
 int sound_free(sound_t *sound)
 {
+    if (sound == NULL){ // error check, nothing passed into function
```
Comment (ecoble) [inline src/sound/src/sound.c]: Style

File: src/sound/src/sound.c [hunk 5]
```diff
@@ -5,17 +5,62 @@
...
     return 0;
 }
 
 /* See sound.h for details */
 int play_sound(sound_t *sound, int delay)
 {
+    if (sound == NULL) { // error check, nothing passed into function
+        fprintf(stderr, "no sound objects present\n");
+        exit(1);
+    }else if (sound->type == BACKGROUND) {
```
Comment (ecoble) [inline src/sound/src/sound.c]: Style

File: src/sound/src/sound.c [hunk 6]
```diff
@@ -5,17 +5,62 @@
...
+        SDL_Delay(delay); // do a delay if there is one
+        Mix_PlayMusic(sound->name, -1); // play the music
+        return 1;
+    } 
+    else if (sound->type == SOUND_EFFECT){
```
Comment (ecoble) [inline src/sound/src/sound.c]: Style

**Review by ecoble — CHANGES_REQUESTED** [review]
This looks good besides some minor style issues!

File: src/sound/sandbox/README [hunk 7]
```diff
@@ -1,42 +1,102 @@
...
+This should build a demo program that opens a window. Once the window opens, background music will play. Three sample sound files are in the same folder as this file. Different key inputs will play different sounds:
+
+`q`: the first wav file in the chunk array is played.
+`w`: the second wav file in the chunk array is played. Note that the wav file started by `q` won't stop.
+`e`: the first mp3 file in the music array is played.
+` `(space): the music is paused.
+`r`: the music is resumed.
+
+
+## Major Drawbacks
```
Comment (ecoble) [inline src/sound/sandbox/README:59]: This is really useful, great documentation!

**Review by ecoble — APPROVED** [review]
Great job!

**Comment by JaStarbuck** [comment]
Closing statement: the three functions, load and play and free, have been created and are not giving any errors during the building process. Also, the file now adheres to the style guide. There are some warnings, however, and they will be updated in a new issue.

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Nice job finishing the sound module, and also great documentation!

**Pull request merged**